### PR TITLE
Refactor factory locator dependency

### PR DESCRIPTION
**Pull request by lucascorbeaux**
Hi Abdul Malik,

Here is the PR requested, with PHPUnit test update too.

We'll see if it works :)

### CODE REVIEW

**Comment by samsonasik** [comment]
Hi Lucas, thank you for the pr. scrutinizer has 2 new issues, see https://scrutinizer-ci.com/g/samsonasik/SanSessionToolbar/inspections/44f8376b-0392-476c-b66e-81675bc3b729/issues/files/src/Factory/Controller/SessionToolbarControllerFactory.php?status=new&orderField=path&order=asc . Travis failure on @cover annotation need update

**Comment by lucascorbeaux** [comment]
Hi Abdul Malik,

I fixed the @cover problem, sorry for the mess I'm not accustomed to using
@cover annotation and didn't seen it.

I ultimately try a new improvement for removing one of the 2 scrutinizer
issue. I don't understand why the second one is marked as new as it was
introduced 11 days ago, and I don't see at all how we can remove it.

Hope this will works this time.

2015-04-06 13:49 GMT+02:00 Abdul Malik Ikhsan <notifications@github.com>:

File: src/Factory/Controller/SessionToolbarControllerFactory.php [hunk 1]
```diff
@@ -20,22 +20,33 @@
...
-        $services = $controllerManager->getServiceLocator();
+        $services = $this->getInnerServiceLocator($serviceLocator);
 
         return new SessionToolbarController(
             $services->get('ViewRenderer'),
             (object) $services->get('SanSessionManager')
         );
     }
+
+    protected function getInnerServiceLocator(ServiceLocatorInterface $serviceLocator)
```
Comment (samsonasik) [inline src/Factory/Controller/SessionToolbarControllerFactory.php]: I prefer to use private method as we won't extend its class. and change name with <code>getParentServiceLocator()</code>. this method can be included in coverage with following alternative : 
1. add more @covers in test.
2. use ReflectionMethod
tests regression can be check with : 
```
phpunit --coverage-text
```

File: test/Factory/Controller/SessionToolbarControllerFactoryTest.php [hunk 2]
```diff
@@ -59,23 +59,36 @@ protected function setUp()
...
+        $this->doTestCreateService($this->controllerManager);
+    }
+
+    /**
+     * @covers SanSessionToolbar\Factory\Controller\SessionToolbarControllerFactory::createService
+     */
+    public function testCreateServiceWithServiceLocator()
+    {
+        $this->doTestCreateService($this->serviceLocator);
+    }
```
Comment (samsonasik) [inline test/Factory/Controller/SessionToolbarControllerFactoryTest.php]: I prefer to use @dataProvider for this.

**Comment by lucascorbeaux** [comment]
Thanks for the coverage check info. I'll look at this ASAP.

The dataProvider were my first try, but it leads to many tricky situations because we're using class properties that are initialized after the dataProvider is challenged. It's doable, but it greatly complexify the test case. That's why I opted for this protected method that don't satisfy me either.

**Comment by samsonasik** [comment]
Thank you!, I've updated my blog post with your alternative suggestion.

**Comment by samsonasik** [comment]
@lucascorbeaux any update for this ? or is it ok to me to take care the coverage ?

**Comment by lucascorbeaux** [comment]
@samsonasik sorry, I have emergencies these days. I plan to do this in a
couple days but if it's simpler for you it's OK for me that you take care
of this.

2015-04-09 11:45 GMT+02:00 Abdul Malik Ikhsan <notifications@github.com>:

**Comment by samsonasik** [comment]
Ok, I will try to take care of that today during merge. Thank you ;)

Warm regards,

Abdul Malik Ikhsan

Pada 9 Apr 2015, pukul 16.49, Lucas CORBEAUX <notifications@github.com> menulis:

**Comment by samsonasik** [comment]
@lucascorbeaux thank you for the contribution ;). It has been merged. I've taken care the coverage during merge ;)

**Comment by lucascorbeaux** [comment]
@samsonasik thank you for the merge, and sorry for not being able to do this quickly myself :)

**Comment by samsonasik** [comment]
you're welcome. it's ok ;)

**Pull request merged**